metis-normalization/src/main/java/eu/europeana/normalization/dates/extraction/dateextractors/CenturyDateExtractor.java

```diff
@@ -0,0 +1,125 @@
+
+import static java.util.regex.Pattern.CASE_INSENSITIVE;
+import static java.util.regex.Pattern.compile;
+
+import eu.europeana.normalization.dates.DateNormalizationExtractorMatchId;
+import eu.europeana.normalization.dates.DateNormalizationResult;
+import eu.europeana.normalization.dates.YearPrecision;
+import eu.europeana.normalization.dates.edtf.AbstractEdtfDate;
+import eu.europeana.normalization.dates.edtf.EdtfDatePart;
+import eu.europeana.normalization.dates.edtf.InstantEdtfDate;
+import eu.europeana.normalization.dates.edtf.IntervalEdtfDate;
+import eu.europeana.normalization.dates.extraction.RomanToNumber;
+import java.util.Arrays;
+import java.util.Objects;
+import java.util.function.ToIntFunction;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * Extractor that matches a century with a decimal or Roman numerals
+ * <p>The range of values this accepts are from 1-21 including.</p>
+ * <p>Examples of some cases:
+ * <ul>
+ *   <li>
+ *     Value = 18.. | Outcome = 18XX
+ *     Value = 1st century | Outcome = 00XX
+ *     Value = s. XXI | Outcome = 20XX
+ *     Value = s.II-III | Outcome = 01XX/02XX
+ *   </li>
+ * </ul>
+ * </p>
+ * <p>The Roman numerals may also be preceded by an abbreviation of century, for example ‘s. XIX’.</p>
+ * <p>Also supports ranges.</p>
+ */
+public class CenturyDateExtractor implements DateExtractor {
+
+  private static final String NUMERIC_10_TO_21_ENDING_DOTS_REGEX = "(1\\d|2[0-1])\\.{2}";
+  private static final String NUMERIC_1_TO_21_SUFFIXED_REGEX = "(2?1st|2nd|3rd|(?:1\\d|[4-9]|20)th)\\scentury";
+  private static final String ROMAN_1_TO_21_REGEX = "(X?(?:IX|IV|VI{0,3}|I{1,3})|X|XXI?)";
+  private static final String CENTURY_PREFIX = "(?:(?:s|sec|saec)\\s|(?:s|sec|saec)\\.\\s?)?";
+  private static final String QUESTION_MARK = "\\??";
+
+  enum PatternCenturyDateOperation {
+    PATTERN_YYYY(
+        compile(QUESTION_MARK + NUMERIC_10_TO_21_ENDING_DOTS_REGEX + QUESTION_MARK, CASE_INSENSITIVE),
+        century -> Integer.parseInt(century) * 100, DateNormalizationExtractorMatchId.CENTURY_NUMERIC),
+    PATTERN_ENGLISH(
+        compile(QUESTION_MARK + NUMERIC_1_TO_21_SUFFIXED_REGEX + QUESTION_MARK, CASE_INSENSITIVE),
+        century -> (Integer.parseInt(century.substring(0, century.length() - 2)) - 1) * 100,
+        DateNormalizationExtractorMatchId.CENTURY_NUMERIC),
+    PATTERN_ROMAN(
+        compile(QUESTION_MARK + CENTURY_PREFIX + ROMAN_1_TO_21_REGEX + QUESTION_MARK, CASE_INSENSITIVE),
+        century -> (RomanToNumber.romanToDecimal(century) - 1) * 100,
+        DateNormalizationExtractorMatchId.CENTURY_ROMAN),
+    PATTERN_ROMAN_RANGE(
+        compile(QUESTION_MARK + CENTURY_PREFIX + ROMAN_1_TO_21_REGEX + "\\s?-\\s?" + ROMAN_1_TO_21_REGEX + QUESTION_MARK,
+            CASE_INSENSITIVE), century -> (RomanToNumber.romanToDecimal(century) - 1) * 100,
+        DateNormalizationExtractorMatchId.CENTURY_RANGE_ROMAN);
+
```

Comment:
NUMERIC_10_TO_21_ENDING_DOTS_REGEX
![image](https://user-images.githubusercontent.com/19560266/190131188-a0a7a071-1ece-4feb-b13d-13827a0e0c53.png)
NUMERIC_1_TO_21_SUFFIXED_REGEX
![image](https://user-images.githubusercontent.com/19560266/190131323-861ffe15-c88a-4bea-9c01-657674cffb6c.png)
ROMAN_1_TO_21_REGEX
![image](https://user-images.githubusercontent.com/19560266/190131877-240d94b5-3702-4d19-9f65-0af8bd709d30.png)
The last values, e.g. XXII, or VVV  are not accepted as expected
CENTURY_PREFIX
![image](https://user-images.githubusercontent.com/19560266/190131966-e47ba26b-35f2-45fd-857a-989fd2bf470c.png)

Reply:
The regex were tested.

Reply:
Lower cases should be supported and the are from the CASE_INSENSITIVE option that is already present.
I've added more unit tests to capture better the option.